platform/LinkedList.h

```diff
@@ -0,0 +1,60 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_LINKED_LIST_H
+#define MBED_LINKED_LIST_H
+
+#include "LinkEntry.h"
+#include "LinkedListBase.h"
+
+template<class T>
+class LinkedList: public LinkedListBase {
```

Comment:
I'd suggest to have the LinkedList part in its own PR and rename it into IntrusiveList to make it clear that the data structure use pointers present in its elements.